provider/netperf.py

```diff
@@ -0,0 +1,83 @@
+import logging
+import os
+
+from virttest import data_dir
+from virttest import utils_misc
+from virttest import utils_net
+from virttest import utils_netperf
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+class NetperfTest(object):
+
+    def __init__(self, params, vm):
+        self.params = params
+        self.vm = vm
+        self.client = self.netperf_client()
+        self.server = self.netperf_server()
+
+    def netperf_client(self):
+        """
+        Init netperf client setup
+        """
+        self.client = utils_netperf.NetperfClient(
+            self.vm.get_address(),
+            self.params.get("client_path"),
+            netperf_source=os.path.join(data_dir.get_deps_dir("netperf"),
+                                        self.params.get("netperf_client_link")),
+            client=self.params.get("shell_client"),
+            port=self.params.get("shell_port"),
+            username=self.params.get("username"),
+            password=self.params.get("password"),
+            prompt=self.params.get("shell_prompt"),
+            linesep=self.params.get("shell_linesep", "\n").encode().decode(
+                'unicode_escape'),
+            status_test_command=self.params.get("status_test_command", ""),
+            compile_option=self.params.get("compile_option", ""))
+        return self.client
+
+    def netperf_server(self):
+        """
+        Init netperf server setup
+        """
+        self.server = utils_netperf.NetperfServer(
+            utils_net.get_host_ip_address(self.params),
+            self.params.get("server_path", "/var/tmp"),
+            netperf_source=os.path.join(data_dir.get_deps_dir("netperf"),
+                                        self.params.get("netperf_server_link")),
+            password=self.params.get("hostpassword"),
+            compile_option=self.params.get("compile_option", ""))
+        return self.server
+
+    def start_netperf_test(self):
+        """
+        Start netperf test between client and server
+        """
+        self.server.start()
+        test_duration = self.params.get_numeric("netperf_test_duration", 120)
+        netperf_output_unit = self.params.get("netperf_output_unit")
+        test_option = self.params.get("test_option", "")
```

Comment:
Seems you have two part of the option definition, below from `test_protocols` to `netperf_output_unit` are sepecifically defined, and `test_option` seems more like "extra_options", what do you think to name it `extra_options`, and better with prefix to avoid name conflict with other tests, so `netperf_extra_options`.